project_recalculate/models/project_project.py

```diff
@@ -0,0 +1,100 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+#
+#    OpenERP, Open Source Management Solution
+#    This module copyright :
+#        (c) 2014 Antiun Ingenieria, SL (Madrid, Spain, http://www.antiun.com)
+#                 Endika Iglesias <endikaig@antiun.com>
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import models, fields
+from openerp.exceptions import Warning  # , RedirectWarning
+from openerp.tools.translate import _
+
+
+class ProjectProject(models.Model):
+    _inherit = 'project.project'
+
+    calculation_type = fields.Selection(
+        [('date_begin', 'Date begin'), ('date_end', 'Date end')],
+        default='date_begin', string='Calculation type',
+        help='how to calculate tasks with date start or date end references')
+
+    def calcule_date_start_date_end(self):
+        vals = {}
+        project_task_obj = self.env['project.task']
+        project_task_ids = project_task_obj.search(
+            [('project_id', '=', self.id)])
```

Comment:
`self` is a recordset, so you're best looping it.
`task_ids` is already available from self, so no need to use a `project_task_obj`:

    for project in self:
        ...
        for task in project.task_ids: